src/algebra/category/CommRing/adjunctions.lean

```diff
@@ -28,8 +28,8 @@ private def free_obj (α : Type u) : CommRing.{u} := ⟨mv_polynomial α ℤ⟩
 def free : Type u ⥤ CommRing.{u} :=
 { obj := free_obj,
   map := λ _ _ f, @ring_hom.of _ _ _ _ (rename f) (rename.is_semiring_hom f),
-  map_id' := λ X, ring_hom.ext _ _ $ funext $ rename_id,
-  map_comp' := λ X Y Z f g, ring_hom.ext _ _ $ funext $ λ p, (rename_rename f g p).symm }
+  map_id' := λ X, ring_hom.ext (funext rename_id),
```

Comment:
@semorrison Do you know why `tidy` can't take care of these?

Reply:
It can. Yury added the explicit proofs, presumably because this definition is painfully slow to elaborate. (My suspicion is just by the end of `mv_polynomial.lean`, the simp set has grown so unwieldy it's hard to use simp at all.)

Reply:
@jcommelin AFAIR, something failed here (probably, `ring_hom.of`), and I wrote everything explicitly.